### PR TITLE
Some scripts to bring up a cromwell server w/ mysql using docker compose

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

**Comment by kcibul** [comment]
Looks like the data in mysql will be lost when the docker compose is shut 'down'.  Is that true?  If so, it would be good to document (or make it so) to mount in a volume in the mysql docker compose so the database survives a restart

**Comment by Horneth** [comment]
Yes that's true. However you can `stop` and then `start`, or `restart` the container and keep your data.

**Comment by kcibul** [comment]
huh -- TIL, who knew!  Very nice!  Might be worth a sentence in the README about that if we get the question in the future

File: scripts/docker-compose-mysql/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Cromwell server on MySQL Database
+
+Uses docker-compose to link together a cromwell docker image (built locally with `sbt docker` or available on [dockerhub](https://hub.docker.com/r/broadinstitute/cromwell/)) with a MySQL docker image.
+To change the version of cromwell used, change the tag in `compose/cromwell/Dockerfile`
+
+## Local
+
+`docker-compose up` from this directory will start a cromwell server running on a mysql instance with local backend.
+The default configuration file used can be found at `compose/cromwell/app-config/application.conf`
+To override it, simply mount a volume containing your custom `application.conf` to `/app-config` (see the Jes section below)
```
Comment (geoffjentry) [inline scripts/docker-compose-mysql/README.md]: This says to see the jes section but I don't see what in there is providing extra info

File: scripts/docker-compose-mysql/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+## Jes
```
Comment (geoffjentry) [inline scripts/docker-compose-mysql/README.md]: JES

Comment (mcovarr) [inline scripts/docker-compose-mysql/README.md]: Pipelines API, really...

File: scripts/docker-compose-mysql/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  cromwell:
+    build: 
+      context: ./compose/cromwell
+    volumes:
+      - ./cromwell-executions:/cromwell-working-dir/cromwell-executions
+    command: ["/wait-for-it/wait-for-it.sh mysql-db:3306 -t 120 -- java -Dconfig.file=/app-config/application.conf -jar /app/cromwell.jar server"]
+    links:
+      - mysql-db
+    ports:
+      - "8800:8000"
```
Comment (geoffjentry) [inline scripts/docker-compose-mysql/docker-compose.yml]: Allowing 80 is useful as it's way easier to use port 80 on GCS VMs if you can get away with it

File: scripts/docker-compose-mysql/jes-cromwell/jes-config/application.conf [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                auth = "application-default"
+              }
+            }
+          }
+    }
+  }
+}
+
+database {
+  db.url = "jdbc:mysql://mysql-db/cromwell_db?useSSL=false"
```
Comment (geoffjentry) [inline scripts/docker-compose-mysql/jes-cromwell/jes-config/application.conf]: Can you add `?rewriteBatchedStatements=true` to this? It won't do anything at the moment but future us will thank us for having it in there now

**Comment by geoffjentry** [comment]
👍  This is sweet

File: scripts/docker-compose-mysql/README.md (hunk 2 above)
Comment (geoffjentry) [inline scripts/docker-compose-mysql/README.md]: Well yeah, but that's a bigger change ;)

**Comment by Horneth** [comment]
@kcibul I added a volume entry to mount the mysql data directory onto the host so the data survives a `docker-compose down`

File: scripts/docker-compose-mysql/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# Cromwell server on MySQL Database
+
+Uses docker-compose to link together a cromwell docker image (built locally with `sbt docker` or available on [dockerhub](https://hub.docker.com/r/broadinstitute/cromwell/)) with a MySQL docker image.
```
Comment (mcovarr) [inline scripts/docker-compose-mysql/README.md]: Should be `Cromwell` and `MySQL` throughout

File: scripts/docker-compose-mysql/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+## Local
+
+`docker-compose up` from this directory will start a cromwell server running on a mysql instance with local backend.
+The default configuration file used can be found at `compose/cromwell/app-config/application.conf`
+To override it, simply mount a volume containing your custom `application.conf` to `/app-config` (see `jes-cromwell/docker-compose.yml` for an example)
+
+## JES
+
+The `jes-cromwell` directory is an example of how to customize the original compose file with a configuration file and environment variables.
+It uses the application default credentials of the host machine. To use it make sure your gcloud is up to date and that your [application-default credentials](https://developers.google.com/identity/protocols/application-default-credentials) are setup.
```
Comment (mcovarr) [inline scripts/docker-compose-mysql/README.md]: setup => set up

File: scripts/docker-compose-mysql/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+Then run `docker-compose -f docker-compose.yml -f jes-cromwell/docker-compose.yml up` to start a cromwell server with a Jes backend on MySQL.
```
Comment (mcovarr) [inline scripts/docker-compose-mysql/README.md]: Jes => JES

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**